rascal2/widgets/project.py

```diff
@@ -206,14 +226,388 @@ def show_project_view(self) -> None:
+        if (
+            orientation == QtCore.Qt.Orientation.Horizontal
+            and role == QtCore.Qt.ItemDataRole.DisplayRole
+            and section != 0
+        ):
+            return self.headers[section - 1].replace("_", " ")
+        return None
+
+    def append_item(self):
+        """Append an item to the ClassList."""
+        self.classlist.append(self.item_type())
+        self.endResetModel()
+
+    def delete_item(self, row: int):
+        """Delete an item in the ClassList.
+
+        Parameters
+        ----------
+        row : int
+            The row containing the item to delete.
+
+        """
+        self.classlist.pop(row)
+        self.endResetModel()
+
+    def index_header(self, index):
+        """Get the header for an index.
+
+        Parameters:
+        -----------
+        index : QModelIndex
+            The model index for the header.
+
+        Returns
+        -------
+        str or None
+            Either the name of the header, or None if none exists.
+
+        """
+        col = index.column()
+        if col == 0:
+            return None
+        return self.headers[col - 1]
+
+
+class ParametersModel(ClassListModel):
+    """Classlist model for Parameters."""
+
+    def __init__(self, classlist: RATapi.ClassList, parent: QtWidgets.QWidget):
+        super().__init__(classlist, parent)
+
+        self.protected_indices = []
+        if self.item_type is RATapi.models.Parameter:
+            for i, item in enumerate(classlist):
+                if isinstance(item, RATapi.models.ProtectedParameter):
+                    self.protected_indices.append(i)
+
+    def flags(self, index):
+        flags = super().flags(index)
+        header = self.index_header(index)
+        # disable mu, sigma if prior type is not Gaussian
+        if self.classlist[index.row()].prior_type != "gaussian" and header in ["mu", "sigma"]:
+            return QtCore.Qt.ItemFlag.NoItemFlags
+        # never allow name editing for protected parameters, allow everything else to be edited by default
+        if header == "fit":
+            flags |= QtCore.Qt.ItemFlag.ItemIsUserCheckable
+        elif header != "name" or (self.edit_mode and index.row() not in self.protected_indices):
+            flags |= QtCore.Qt.ItemFlag.ItemIsEditable
+
+        return flags
+
+
+class ProjectFieldWidget(QtWidgets.QWidget):
+    """Widget to show a project ClassList."""
+
+    classlist_model = ClassListModel
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.table = QtWidgets.QTableView(parent)
+        self.table.setSizePolicy(
+            QtWidgets.QSizePolicy.Policy.MinimumExpanding, QtWidgets.QSizePolicy.Policy.MinimumExpanding
+        )
+
+        layout = QtWidgets.QVBoxLayout()
+        # change to icon: remember to mention that plus.png in the icons is wonky
+        self.add_button = QtWidgets.QPushButton("+")
```

Comment:
what do you mean by 'with all the scroll bars'? the list of parameters should start scrolling but the + button should remain fixed at the bottom of it, outside of the scroll area?

Reply:
Sorry not very clear, I meant 

scrolling is required to see the add button which can be confusing since you cannot see the header to know which table the buttons belongs to